cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
+        """
+        :return: A symbolic representation of the hidden features
+        :param layer: The name of the hidden layer to return features at.
+        """
+        error = 'Feature extraction for hidden layers not implemented'
+        raise NotImplementedError(error)
+
+    def get_logits(self):
```

Comment:
is there a strong reason to special-case the logits? Why not just `get_hidden('logits')`?